core/src/main/java/org/apache/iceberg/actions/SizeBasedFileRewriter.java

```diff
@@ -111,6 +111,8 @@ public abstract class SizeBasedFileRewriter<T extends ContentScanTask<F>, F exte
   private boolean rewriteAll;
   private long maxGroupSize;
 
+  private int outputSpecId;
```

Comment:
Why not also save the outputSpec as well? I notice we just have to get it a few times. Maybe it just makes sense as a protected method 
```java
outputSpec() {
  return table.specs().get(outputSpecId)
}
```

Reply:
make sense, will add it.